classes/choice_importer.php

```diff
@@ -173,6 +173,13 @@ public function import($content, $live=true) {
                                 continue;
                             }
 
+                            if ($fieldname === 'title' && strlen($cell) > 255) {
+                                $importstatus->status = self::IMPORT_STATUS_DATA_ERROR;
+                                $importstatus->errors[] = get_string('csvupload_too_long_title',
+                                        RATINGALLOCATE_MOD_NAME, $cell);
+                                break;
```

Comment:
This break only exits the loop for mapping cell content to field names. This leads to creating choices which do not have a title if its too long. But a choice shouldn't be created in these cases. Hence, there should be some form of check whether an error is present when creating them in lines 211-214.

Reply:
Yes, good catch. I might have been a bit fast there, sorry.

I now changed the behavior so that other choices are being properly imported, only the one with the too long title is being skipped and an error is thrown. I am not sure, if this is ideal. Would you suggest to cancel the whole import?

Reply:
Thanks! I also thought about it, I think it's more intuitive if we keep it like this and already import the other choices. Although I noticed that you can import the same choice twice. It would make more sense if the title is unique, however I'm not sure whether there exists some corner scenarios for which teachers want to create separate choices with the same name. I'll create an issue for that.

Reply:
Yes that sounds like a nice feature, feel free to add an issue for that :)